service/history/task/transfer_active_task_executor.go

```diff
@@ -195,6 +196,10 @@ func (t *transferActiveTaskExecutor) processActivityTask(
 	// release the context lock since we no longer need mutable state builder and
 	// the rest of logic is making RPC call, which takes time.
 	release(nil)
+
+	// Ratelimiting is not done. This is only to count the number of requests via metrics
+	t.wfIDCache.AllowInternal(task.DomainID, task.WorkflowID)
```

Comment:
we need this in timer active task executor as well

Reply:
It is not needed for timer tasks because we dont plan to rate limit timer tasks